src/FacadeVFS.js

```diff
@@ -215,6 +215,46 @@ export class FacadeVFS extends VFS.Base {
     return this.jOpen(filename, pFile, flags, pOutFlagsView);
   }
 
+  /**
+   * @param {number} pVfs 
+   * @param {number} nByte 
+   * @param {number} pCharOut
+   * @returns {number|Promise<number>}
+   */
+  xRandomness(pVfs, nByte, pCharOut) {
+    const randomArray = new Uint8Array(nByte);
+    crypto.getRandomValues(randomArray);
+    // Copy randomArray to the WebAssembly memory
+    const buffer = pCharOut; // Pointer to memory in WebAssembly
+    this._module.HEAPU8.set(randomArray, buffer); // Copy randomArray into memory starting at buffer
+    return nByte;
+  }
+
+   /**
+   * Gets the current time as milliseconds since Unix epoch
+   * @param {number} pVfs pointer to the VFS
+   * @param {number} pTime pointer to write the time value
+   * @returns {number} SQLite error code
+   */
+    xCurrentTimeInt64(pVfs, pTime) {
+     // Create a DataView to write the current time
+     const timeView = this.#makeTypedDataView('BigInt64', pTime);
+      
+     // Exact copy of SQLite's calculation:
+     // static const sqlite3_int64 unixEpoch = 24405875*(sqlite3_int64)8640000;
+     // *piNow = unixEpoch + 1000*(sqlite3_int64)sNow.tv_sec + sNow.tv_usec/1000;
+     const unixEpoch = 24405875n * 8640000n;
+     const seconds = BigInt(Math.floor(Date.now() / 1000));
+     const microseconds = BigInt(Date.now() % 1000) * 1000n;
+     
+     const value = unixEpoch + (1000n * seconds) + (microseconds / 1000n);
```

Comment:
I might be confused by the logic here. The method comment mentions "Gets the current time as milliseconds since Unix epoch" 

Could this not be simplified by returning `Date.now()` which according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now) returns `The Date.now() static method returns the number of milliseconds elapsed since the [epoch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date)`

Reply:
```
Find the current time (in Universal Coordinated Time).  Write into *piNow the current time and date as a Julian Day number times 86_400_000.  In other words, write into *piNow the number of milliseconds since the Julian epoch of noon in Greenwich on November 24, 4714 B.C according to the proleptic Gregorian calendar.

static int unixCurrentTimeInt64(sqlite3_vfs *NotUsed, sqlite3_int64 *piNow){
...
}
```
This is how sqlite3 calculates the currentTime, I used the same formula and values to calculate it in the VFS.